main.go

```diff
@@ -94,16 +95,21 @@ func (s *awsSQSSink) Sink(ctx context.Context, datumStreamCh <-chan sinksdk.Datu
 	})
 	if err != nil {
 		s.logger.Errorf("failed to push message %v", err)
-	} else {
-		// check for response in case of partial failure, then log those responses otherwise return id of success request.
-		if len(response.Failed) > 0 {
-			s.logger.Error("failed to push message, err: %v", response.Failed)
-		}
-		if len(response.Successful) > 0 {
-			for _, success := range response.Successful {
-				ok = ok.Append(sinksdk.ResponseOK(aws.ToString(success.Id)))
-			}
+	}
+
+	// log the failure response and return the ID of that.
+	if len(response.Failed) > 0 {
+		s.logger.Error("failed to push message, err: %v", response.Failed)
+
+		for _, fail := range response.Failed {
+			failed = failed.Append(sinksdk.ResponseFailure(aws.ToString(fail.Id), "failed to "))
 		}
+
+		return failed
+	}
+
+	for _, success := range response.Successful {
+		ok = ok.Append(sinksdk.ResponseOK(aws.ToString(success.Id)))
```

Comment:
I think we are returning all the responses, including both failed and successful ones.

Reply:
Agree, that would be better. I'll make the change